thorlcr/mfilemanager/thmfilemanager.cpp

```diff
@@ -145,7 +145,47 @@ class CFileManager : public CSimpleInterface, implements IThorFileManager
         return max;
     }
 
-    void remove(IDistributedFile &file, IUserDescriptor &userDesc)
+    template<typename T>
+    T blockReportFunc(CJobBase &job, std::function<T(unsigned timeout)> func, unsigned timeout, const char *msg)
+    {
+        // every minute timeout, and set workunit state to blocked for NN minutes
+        WUState wuidState = WUStateUnknown;
+        CTimeMon tm(timeout);
+        if (timeout >= 60000)
+            timeout = 60000;
+        while (true)
+        {
+            try
+            {
+                T ret = func(timeout);
+                if (WUStateUnknown != wuidState)
+                {
+                    Owned<IWorkUnit> workunit = &job.queryWorkUnit().lock();
+                    workunit->setState(wuidState);
+                }
+                return ret;
+            }
+            catch (IDFS_Exception *e)
+            {
+                if (DFSERR_LookupConnectionTimout == e->errorCode())
+                {
+                    if (tm.timedout())
+                        throw;
+                }
+                e->Release();
+            }
+            Owned<IWorkUnit> workunit = &job.queryWorkUnit().lock();
+            if (WUStateUnknown == wuidState)
```

Comment:
What if the state is not Unknown at this moment ?

Reply:
it should be.. and I think this condition is ok, but if getState() below were to return WUStateUnknown, it should spot and error out at the point, I'll add a check etc.